src/utils/switchChain.ts

```diff
@@ -2,36 +2,22 @@ import { Connector } from '@web3-react/types'
 import { networkConnection, walletConnectConnection } from 'connection'
 import { getChainInfo } from 'constants/chainInfo'
 import { isSupportedChain, SupportedChainId } from 'constants/chains'
-import { RPC_URLS } from 'constants/networks'
+import { FALLBACK_URLS, RPC_URLS } from 'constants/networks'
 
-function getRpcUrls(chainId: SupportedChainId): [string] {
+function getRpcUrl(chainId: SupportedChainId): string {
   switch (chainId) {
     case SupportedChainId.MAINNET:
     case SupportedChainId.RINKEBY:
     case SupportedChainId.ROPSTEN:
     case SupportedChainId.KOVAN:
     case SupportedChainId.GOERLI:
-      return [RPC_URLS[chainId]]
-    case SupportedChainId.OPTIMISM:
-      return ['https://mainnet.optimism.io']
-    case SupportedChainId.OPTIMISTIC_KOVAN:
-      return ['https://kovan.optimism.io']
-    case SupportedChainId.ARBITRUM_ONE:
-      return ['https://arb1.arbitrum.io/rpc']
-    case SupportedChainId.ARBITRUM_RINKEBY:
-      return ['https://rinkeby.arbitrum.io/rpc']
-    case SupportedChainId.POLYGON:
-      return ['https://polygon-rpc.com/']
-    case SupportedChainId.POLYGON_MUMBAI:
-      return ['https://rpc-endpoints.superfluid.dev/mumbai']
-    case SupportedChainId.CELO:
-      return ['https://forno.celo.org']
-    case SupportedChainId.CELO_ALFAJORES:
-      return ['https://alfajores-forno.celo-testnet.org']
+      return RPC_URLS[chainId][0]
+    // Attempting to add a chain using an infura URL will not work, as the URL will be unreachable from the MetaMask background page.
+    // MetaMask allows switching to any publicly reachable URL, but for novel chains, it will display a warning if it is not on the "Safe" list.
```

Comment:
I would explain this comment within the fallback urls definition instead of here maybe?

Reply:
There's a longer explanation (with referenced source code) in the FALLBACK_URLS definition - this is here too so it's apparent why RPC_URLS is invalid in these cases.